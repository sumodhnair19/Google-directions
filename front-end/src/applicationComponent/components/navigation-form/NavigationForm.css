.form-input {
  width: 100%;
  padding: 12px 20px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-sizing: border-box;
  text-align: center;
}

.btn-container>.form-input {
  float: left;
  width: 45%;
  margin: 15px 5px;
}

.form-container {
  border-radius: 5px;
  background-color: #f2f2f2;
  padding: 20px;
  float: left;
  text-align: left;
  width: 20.6em;
}

.parent {
  position: relative;
}

.close {
  position: absolute;
  top: 50%;
  right: -15px;
  border: 1px solid black;
  padding: 1px;
  border-radius: 4px;
  cursor: pointer;
}

@media only screen and (max-width: 1023px) {
  .form-container {
    width: 35%;
  }
}

@media only screen and (max-width: 767px) {
  .form-container {
    width: 95%;
  }
}
