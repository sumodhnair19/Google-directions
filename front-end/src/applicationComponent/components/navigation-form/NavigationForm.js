import React, { Component } from 'react';
import { googleMaps } from '../../../sharedComponent/googleMap';

import './NavigationForm.css';
/**
 * @name NavigationForm
 * @type {Component}
 * @extends Component React component
 * @description This component contains the auto complete form
 */

class NavigationForm extends Component {

    fromInput;   //starting pt input
    fromInputAutoComplete;  //google autocomplete starting pt will render ref in this
    toInputAutoComplete;   //google autocomplete ending pt will render ref in this
    toInput;  //ending pt input

    constructor() {
      super();

      this.state = {
        fromInput: this.fromInput,
        toInput : this.toInput
      }
    }
    /**
     * @name getRoute
     * @description call the parent 'getDirections' method and supply current Starting and drop-off points
     */

    getRoute = () => {
        let from = this.fromInputAutoComplete.getPlace(),
            to = this.toInputAutoComplete.getPlace();
        if(!from || !to || !this.fromInput.value || !this.toInput.value)
            return;

          this.props.getDirections(from, to);
    };
    /**
       * @name renderData
       * @description Attaches the google places autocomplete to the inputs
       */
    renderData = async () => {
        const googleMaps = await this.props.googleMaps();

        this.fromInputAutoComplete = new googleMaps.places.Autocomplete(this.fromInput);

        this.toInputAutoComplete = new googleMaps.places.Autocomplete(this.toInput);
    };
    /**
       * @name resetRoute
       * @description resets the inputs to default
       */
    resetRoute = () =>  {
      this.fromInput.value = "";
      this.toInput.value= "";
      this.setState({
        fromInput: this.fromInput,
        toInput : this.toInput
      })
    }


    componentDidMount() {
        this.renderData();
    }

    resetInput = (val) => {
      if(val) {
        if(val==='from')
           this.fromInput.value = "";
       else
           this.toInput.value = "";
        this.setState({
          fromInput: this.fromInput,
          toInput : this.toInput
        })
      }


    }
    render() {
        return (
            <div className="form-container">
                <div className="parent">
                    <label>Starting Location</label>
                    <input type="text" className="form-input" ref={el => (this.fromInput = el)} />
                      <span className="close" onClick={this.resetInput.bind(this,'from')}>X</span>
                </div>
                <div className="parent">
                    <label>Drop-off point</label>
                    <input type="text"  className="form-input" ref={el => (this.toInput = el)}/>
                    <span className="close" onClick={this.resetInput.bind(this,'to')}>X</span>
                </div>
                <div className="btn-container">
                    <button className="form-input" onClick={this.getRoute.bind(this)}>Submit</button>
                    <button className="form-input" onClick={this.resetRoute.bind(this)}>Reset</button>
                </div>
            </div>
        );
    }
}

NavigationForm.defaultProps = {
    googleMaps
};

export default NavigationForm;
