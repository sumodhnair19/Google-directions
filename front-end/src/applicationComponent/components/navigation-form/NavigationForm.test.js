import React from 'react';
import renderer from 'react-test-renderer';

import NavigationForm from './NavigationForm';

describe('Navigation Form Test', () => {
    it('render correctly', () => {
        const tree = renderer.create(<NavigationForm />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
