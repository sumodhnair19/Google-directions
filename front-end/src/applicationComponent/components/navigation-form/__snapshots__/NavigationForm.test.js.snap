// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Form Test render correctly 1`] = `
<div
  className="form-container"
>
  <div>
    <label>
      Starting Location
    </label>
    <input
      className="form-input"
      type="text"
    />
  </div>
  <div>
    <label>
      Drop-off point
    </label>
    <input
      className="form-input"
      type="text"
    />
  </div>
  <div
    className="btn-container"
  >
    <button
      className="form-input"
      onClick={[Function]}
    >
      Submit
    </button>
    <button
      className="form-input"
      onClick={[Function]}
    >
      Reset
    </button>
  </div>
</div>
`;
